source/validations.ts

```diff
@@ -0,0 +1,123 @@
+// /source/validations.ts
+// The validation functions
+
+import { isIP } from 'node:net'
+import type { Request } from 'express'
+
+/**
+ * An error thrown/returned when a validation error occurs.
+ */
+class ValidationError extends Error {
+	name: string
+	code: string
+
+	/**
+	 * The code must be a string, in snake case and all capital, that starts with
+	 * the substring `ERR_ERL_`.
+	 *
+	 * The message must be a string, starting with a lowercase character,
+	 * describing the issue in detail.
+	 */
+	constructor(code: string, message: string) {
+		super(
+			`express-rate-limit: ${code} - ${message} See https://github.com/express-rate-limit/express-rate-limit/wiki/Error-Codes#${code.toLowerCase()} for more information on this error.`,
+		)
+
+		// `this.constructor.name` is the class name
+		this.name = this.constructor.name
+		this.code = code
+		this.message = message
+	}
+}
+
+export class Validations {
+	constructor(private enabled: boolean) {}
+
+	disable() {
+		this.enabled = false
+	}
+
+	/**
+	 * Checks whether the IP address is valid, and that it does not have a port
+	 * number in it.
+	 *
+	 * See https://github.com/express-rate-limit/express-rate-limit/wiki/Error-Codes#err_erl_invalid_ip_address.
+	 *
+	 * @param ip {string | undefined} - The IP address provided by Express as request.ip.
+	 *
+	 * @returns {void}
+	 */
+	ip(ip: string | undefined) {
+		this.wrap(() => {
+			if (ip === undefined) {
+				throw new ValidationError(
+					'ERR_ERL_UNDEFINED_IP_ADDRESS',
+					`An undefined 'request.ip' was detected. This might indicate a misconfiguration or the connection being destroyed prematurely.`,
+				)
+			}
+
+			if (!isIP(ip)) {
```

Comment:
I had thought about using this function, but it also allows port numbers in the IP address so I didn't use it.

Reply:
Oh, does it? I have a couple of tests for that, but maybe I did it wrong. I need to get some sleep, but I'll try and take a closer look tomorrow.

Reply:
Nevermind, seems like it does not allow port numbers, so it works :)